lib/request_handler/validation/dry_backend.rb

```diff
@@ -0,0 +1,56 @@
+require_relative 'errors'
+require_relative 'backend'
+
+module RequestHandler
+  module Validation
+    class DryBackend < Backend
+      def self.valid_schema?(schema)
+        schema.respond_to?(:call)
+      end
+
+      def validate
+        self.validation_result = if options.empty?
+                                   schema.call(value)
+                                 else
+                                   schema.with(options).call(value)
+                                 end
+        self.valid = determine_validity_of_result
+        validation_result
+      rescue Dry::Types::ConstraintError
+        self.valid = false
+        nil
+      end
+
+      def validate!
+        validate.tap do
+          raise Validation::Error unless valid?
+        end
+      end
+
+      def valid?
+        validate if valid.nil?
+        valid
+      end
+
+      def errors
+        validation_result.errors
+      end
+
+      def output
+        validation_result.output
+      end
+
+      private
+
+      attr_accessor :validation_result, :valid
+
+      def determine_validity_of_result
+        if validation_result.respond_to?(:success?)
```

Comment:
did we originally had this logic in this form including fallback to `true` somewhere?

Reply:
yes and no the problem is that sometimes we had this code `Dry::Types::Strict::String.enum(*args).call` and for real schemas we have `(Dry::Validation.Schema {}).call` When you do it with Dry-types it raises an error, but with the schema not. Its a bit weird, will think of a nicer solution